cpp/src/io/statistics/byte_array_view.cuh

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/utilities/span.hpp>
+
+namespace cudf::io {
+
+/**
+ * @brief Wrapper for a row of a list<int8> or list<uint8> column. This is analagous to
+ * `string_view` in type. It was created due to the need for comparison operators for cub reduce on
+ * statistics. Otherwise, it is a device_span in all but name.
+ *
+ */
+class byte_array_view {
```

Comment:
Maybe we should derive from `device_span<uint8_t const>` to inherit the basic members.

Reply:
Alternatively, feels like there could be a base template class for string_view and byte_array_view, data access and comparison semantics seem to be the same.
Maybe these can be future enhancements.

Reply:
We talked about this being derived and the pros/cons above. I debated that, but didn't want to give immediate access to all the members in span. I'm not against it and would encourage the debate and conversions that result, but would suggest a followup for that.

Reply:
Sorry, that conversation was [here](https://github.com/rapidsai/cudf/pull/11303#discussion_r926958393).

Reply:
Yeah, this can be changed later. Mostly wanted to share the ideas (was not aware of the previous convo).

Reply:
Why does this need to be a distinct type instead of just using `device_span` directly?

Reply:
I need the comparison operators and it seemed nice, but it isn't strictly required.

Reply:
> but didn't want to give immediate access to all the members in span. 

Would `private` inheritance work, with some `using` declarations to expose the methods we'd like to?

Reply:
You can always pass a custom comparison operator to an algorithm.

Reply:
It certainly does, but my concern from using a span was that I wanted to abstract away the internals of it and not require the user to know it is a span of `uint8_t`, but they do need to know that for the data pointer, etc so this could be a moot point. I'm not against doing this, but I need to get moving on it as this is the foundation of #11303, #11160, and #11328, which are all 22.08 PRs.

Reply:
@mythrocks 
> Would private inheritance work, with some using declarations to expose the methods we'd like to?

This is nice. Less verbose, chances of logical mistakes will be less compared to composition. Since `using` declaration inherits all overloads, it needs some thought. when overloads of `span` changes or added, so does derived class overloads.

I hope, it's not an anti-pattern (@codereport)